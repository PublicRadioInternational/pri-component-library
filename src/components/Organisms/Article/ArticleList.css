/* import colors... */
@value colors: "../../00_global/colors.css";
@value grayDark, grayLight, whiteDark from colors;

.section {
  background-color: whiteDark;
}

.header {
  padding: 10px 15px;
  font-size: 1rem;
  border-top: 3px solid grayLight;
  color: grayDark;
}

.footer {
  padding: 1rem;
}

.footer a {
  display: block;
  width: 100%;
  font-weight: normal;
  text-transform: none;
  box-sizing: border-box;
}

.footer a::after {
  content: ' >';
}
