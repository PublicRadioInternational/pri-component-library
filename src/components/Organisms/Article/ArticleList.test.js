/**
 * @file ArticleList.test.js
 * Contains tests for ArticleList.component.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import ArticleList from './ArticleList.component';

describe('<ArticleItem />', () => {
  it('Matches the Article List with more link snapshot', () => {
    const component = renderer
      .create(
        <ArticleList
          title="Test Title"
          moreTitle="More Stories"
          moreUrl="more/"
        >
          <span>Test Item One</span>
          <span>test Item Two</span>
        </ArticleList>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('Matches the Article List without more link snapshot', () => {
    const component = renderer
      .create(
        <ArticleList title="Test Title">
          <span>Test Item One</span>
          <span>test Item Two</span>
        </ArticleList>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
