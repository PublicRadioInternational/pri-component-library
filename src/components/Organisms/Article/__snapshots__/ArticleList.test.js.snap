// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleItem /> Matches the Article List with more link snapshot 1`] = `
<section
  className="className && section"
>
  <header
    className="header"
  >
    Test Title
  </header>
  <span>
    Test Item One
  </span>
  <span>
    test Item Two
  </span>
  <footer
    className="footer"
  >
    <a
      className="btnBlue null"
      href="more/"
      onClick={[Function]}
    >
      More Stories
    </a>
  </footer>
</section>
`;

exports[`<ArticleItem /> Matches the Article List without more link snapshot 1`] = `
<section
  className="className && section"
>
  <header
    className="header"
  >
    Test Title
  </header>
  <span>
    Test Item One
  </span>
  <span>
    test Item Two
  </span>
</section>
`;
